### PR TITLE
Move skorch

### PR DESCRIPTION
**Pull request by eukaryo**
## Motivation

Progress https://github.com/optuna/optuna/issues/4484
## Description of the changes

- Move `integration/skorch.py` and its test from `optuna/optuna`.

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@eukaryo Please resolve conflicts.

**Comment by eukaryo** [comment]
@HideakiImamura I have resolved current conflict, and all checks have passed.

File: README.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ Optuna-Integration API reference is [here](https://optuna-integration.readthedoc
 * [Chainer](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#chainer)  ([example](https://github.com/optuna/optuna-examples/tree/main/chainer/chainer_integration.py))
 * [ChainerMN](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#chainermn)  ([example](https://github.com/optuna/optuna-examples/tree/main/chainer/chainermn_simple.py))
 * [Keras](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#keras)  ([example](https://github.com/optuna/optuna-examples/tree/main/keras))
+* [skorch](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#skorch)
```
Comment (HideakiImamura) [inline README.md:24]: An example of `skorch` is [here](https://github.com/optuna/optuna-examples/blob/6d3fedcadc67c3f81b882fad9dffd28612967a0c/pytorch/skorch_simple.py).

File: optuna_integration/skorch.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+from typing import Any
+
+import optuna
+
+
+with optuna._imports.try_import() as _imports:
```
Comment (HideakiImamura) [inline optuna_integration/skorch.py:6]: Please use `optuna-integration` side `_imports`.

File: README.md (hunk 1 above)
Comment (eukaryo) [reply README.md]: I have added the link.

File: optuna_integration/skorch.py (hunk 2 above)
Comment (eukaryo) [reply optuna_integration/skorch.py]: I have fixed it.

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Pull request merged**